consensus/safety-rules/src/tests/local.rs

```diff
@@ -2,17 +2,33 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{test_utils, tests::suite, SafetyRulesManager, TSafetyRules};
+use libra_crypto::{ed25519::Ed25519PrivateKey, Uniform};
 use libra_types::validator_signer::ValidatorSigner;
 
 #[test]
 fn test() {
-    suite::run_test_suite(safety_rules);
+    suite::run_test_suite(safety_rules, false);
+    suite::run_test_suite(safety_rules, true);
 }
 
-fn safety_rules() -> (Box<dyn TSafetyRules>, ValidatorSigner) {
+fn safety_rules(
+    verify_vote: bool,
+) -> (
+    Box<dyn TSafetyRules>,
+    ValidatorSigner,
+    Option<Ed25519PrivateKey>,
+) {
     let signer = ValidatorSigner::from_int(0);
     let storage = test_utils::test_storage(&signer);
-    let safety_rules_manager = SafetyRulesManager::new_local(signer.author(), storage);
+    let safety_rules_manager = SafetyRulesManager::new_local(signer.author(), storage, verify_vote);
     let safety_rules = safety_rules_manager.client();
-    (safety_rules, signer)
+    (
+        safety_rules,
+        signer,
+        if verify_vote {
+            Some(Ed25519PrivateKey::generate_for_testing())
```

Comment:
I mean for tests... why not just always pass in a key

Reply:
What's the difference?